account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -71,8 +74,14 @@ export async function createSMAV2Account(
     initialOwner,
     accountAddress,
     entryPoint = getEntryPoint(chain, { version: "0.7.0" }),
+    isGlobalValidation = true,
+    entityId = 0n,
   } = config;
 
+  if (entityId >= 2n ** 32n) {
```

Comment:
💬 I see validation then wonder if we should have done a parse instead. There are many articles + videos, but it comes from a more type driven programming. https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/
And in this case the phantom types could be the parsed proof https://matheusvellone.medium.com/using-phantom-types-to-extreme-type-guard-bec685d6695c
And all this was just to bring up the topic with an example, not to decide what is right and wrong. I usually just like showing other ways of doing things. Getting something out faster, this is usually the better approach.

Reply:
Love this! Is there a `uint32` type we could use here instead of `bigint`?

Reply:
ah, phantom types, i see. Ok I'll read up and implement this in! (next week, will leave these in for now)

Reply:
Please leave them in for now, I just 
A. Wanted to show the phantom types
B. Get people a new tool 

Because now we have the ability to make an API that reduces the chances of the wrong things fitting into other holes. While not removing the previous tools and API's used previously.

Reply:
In rust this would be a Newtype https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html
But it suffers from more issues compared to TS because TS doesn't care of ordering. There could be some trait type tricks but those also have issues that TS doesn't have.